pilot/pkg/model/sidecar.go

```diff
@@ -357,7 +357,7 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
 
 			for h, ports := range virtualServiceDestinations(v) {
 				// Default to this hostname in our config namespace
-				if s, ok := ps.ServiceIndex.HostnameAndNamespace[host.Name(h)][configNamespace]; ok {
+				if s, ok := ps.ServiceIndex.HostnameAndNamespace[host.Name(h)][configNamespace]; ok && ps.IsServiceVisible(s, configNamespace) {
```

Comment:
I think this may violate original intention, it meant to infer svc from virtual service destination, actually these services are not exported to the sidecar

Reply:
But we do the same filtering in else block

Reply:
@hzxuzhonghu can you PTAL when you get chance?

Reply:
This is unneccessary, service export to `NONE` will not be in SidecarIndex at all.

And here the service is in same namespace as proxy configNamespace, so it must be visible to proxy even it is private.

Reply:
These are services determined by virtual service destination not services in sidecarIndex.  See https://github.com/istio/istio/blob/master/pilot/pkg/model/sidecar.go#L385 - It does similar check. I think we need same check in if block. BTW, we ran in to this issue recently. Service with export to `~` has its clusters and routes created.